dnf5daemon-server/dbus/interfaces/org.rpm.dnf.v0.rpm.Rpm.xml

```diff
@@ -188,6 +188,26 @@ along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
         <arg name="options" type="a{sv}" direction="in" />
     </method>
 
+    <!--
+        transaction_before_begin:
+        @total: number of elements in the rpm transaction
+
+        Send right before the rpm transaction is run.
+    -->
+    <signal name="transaction_before_begin">
+        <arg name="total" type="i" />
```

Comment:
I'd like to see there `type="t"` (unsigned 64-bit integer) rather than `"i"` (signed 32-bit integer). I know that `"i"` directly reflects what `rpmtsNElements()` returns on x86_64. However:

* We already use "t" in `transaction_action_start` and `transaction_action_progress` signals for counting transactions. Mixing signed and unsigned values brings a risk for programs (e.g. in C comparison between the two is undefined.)
* Cautious clients would have to handle negative values. What would a negative value mean? I guess people will use that value for rendering progress bars.
* There could be platforms where C `int` is not 32-bit. 64-bit integer buys us some time.

Could you please consider changing it to `t` type?

Reply:
Yeah, I was thinking about it and eventually went with what rpmtsNElements. But I agree with your signed/unsigned reasoning. I will change it.